addons/website_sale_delivery/controllers/main.py

```diff
@@ -27,6 +27,8 @@ def update_eshop_carrier(self, **post):
         order = request.website.sale_get_order()
         carrier_id = int(post['carrier_id'])
         if order:
+            if any(order.transaction_ids.filtered(lambda tx: tx.state not in ["canceled", "error"])):
```

Comment:
Written as such it is not very efficient, though it mostly likely wont matter, any accepts generators
```suggestion
            if any(tx.state not in ("canceled", "error") for tx in order.transaction_ids):
```